### ISSUE TITLE
'cd' results in "failed assertion"

### ISSUE THREAD

**Issue by rdzman** [octave-app/octave-app#32]
## Changing directories sometimes yields "failed assertion"

If I launch Octave.app (currently using 5.1.0 beta3) and use `cd` to change the directory, I sometimes get a warning like the following:

    2019-04-17 10:18 octave-gui[42972] (FSEvents.framework) FSEventStreamFlushSync(): failed assertion '(SInt64)last_id > 0LL'

The strange part is that it is frequent, but not completely consistent and deterministically reproducible.
* Sometimes typing simply `cd` in a freshly opened app will result in the warning.
* Sometimes cd-ing to another directory results in the warning, sometimes it does not. And it does not seem to be related to the destination directory or where I'm cd-ing from (not consistently repeatable).
* And I see the issue with both the command form (`cd <new_dir>`) and the function call form (`cd('<new_dir>')`).
* Sometimes the warning even displays immediately on launch, before typing any `cd` command.

I discovered the issue by running some automated tests in the GUI that work fine with no warnings when run from an octave session in Terminal. In the GUI they always (or almost always) produce these warnings, and I tracked it down to some calls to `cd`.

Note: I'm often encountering #24 as well, but when I'm able to get a GUI up and running without the spinning beachball, I see the behavior reported here.

## Environment and Diagnostics

I'm running Octave.app 5.1.0 beta3 on a MacBook Pro (Retina, 15-inch, Mid 2014) running macOS High Sierra 10.13.6.

Please run `system env` in Octave.app, and copy & paste the output here:

```
>> system env
SHELL=/bin/bash
TERM=xterm
TMPDIR=/var/folders/jg/9wj7zjkd0xs65bhpww3gphvh0000gn/T/
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.3Puj9jAbwY/Render
LC_NUMERIC=C
USER=ray
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.DjJC07OLnh/Listeners
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
COLUMNS=82
PATH=/Applications/Octave-5.1.0.app/Contents/Resources/usr/bin/:/Applications/Octave-5.1.0.app/Contents/Resources/usr/bin/:/usr/local/sbin:/usr/local/opt/ruby/bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/Users/janke/.rvm/bin:/Applications/Octave-5.1.0.app/Contents/Resources/usr/Cellar/octave-octave-app@5.1.0/5.1.0/libexec/octave/5.1.0/site/exec/x86_64-apple-darwin16.7.0:/Applications/Octave-5.1.0.app/Contents/Resources/usr/Cellar/octave-octave-app@5.1.0/5.1.0/libexec/octave/api-v53/site/exec/x86_64-apple-darwin16.7.0:/Applications/Octave-5.1.0.app/Contents/Resources/usr/Cellar/octave-octave-app@5.1.0/5.1.0/libexec/octave/site/exec/x86_64-apple-darwin16.7.0:/Applications/Octave-5.1.0.app/Contents/Resources/usr/Cellar/octave-octave-app@5.1.0/5.1.0/libexec/octave/5.1.0/exec/x86_64-apple-darwin16.7.0:/Applications/Octave-5.1.0.app/Contents/Resources/usr/Cellar/octave-octave-app@5.1.0/5.1.0/bin
_=/usr/bin/env
PWD=/Users/ray
GS_OPTIONS=-sICCProfilesDir=/Applications/Octave-5.1.0.app/Contents/Resources/usr/opt/ghostscript/share/ghostscript//iccprofiles/ -sGenericResourceDir=/Applications/Octave-5.1.0.app/Contents/Resources/usr/opt/ghostscript/share/ghostscript//Resource/ -sFontResourceDir=/Applications/Octave-5.1.0.app/Contents/Resources/usr/opt/ghostscript/share/ghostscript//Resource/Font
LANG=en_US.UTF-8
XPC_FLAGS=0x0
F77=/Applications/Octave-5.1.0.app/Contents/Resources/usr/bin/gfortran
LINES=98
XPC_SERVICE_NAME=0
HOME=/Users/ray
SHLVL=2
FC=/Applications/Octave-5.1.0.app/Contents/Resources/usr/bin/gfortran
GNUTERM=qt
LOGNAME=ray
DISPLAY=/private/tmp/com.apple.launchd.X3HDckXCle/org.macosforge.xquartz:0
LC_TIME=C
ans = 0
>>
```

If your problem is weird, please also run `octave_app_diagnostic_dump` in Octave.app and attach its output file to this issue.

If your problem is Octave.app freezing with a spinning “beachball” mouse cursor, if you can, please take a spindump, zip it, and attach it to this issue. (To create a spindump, open Activity Monitor while Octave.app is frozen, select the `octave-gui` process, click the “gear” button in the upper left, and choose "Spindump".)

**Comment by apjanke** [comment]
So, I'm _pretty_ sure this is actually a bug in macOS's file-monitoring API, which is being exposed through Qt's FileWatcher classes. This is one of our earliest bugs: https://github.com/octave-app/octave-app-bundler/issues/13. It is disturbing but apparently harmless. It's exposed in the GUI but not in the CLI because it's only desktop GUI components that use this file change notification API, apparently.

We've gone so far as to use a hacked version of Qt that just removes the `FSEventStreamFlushSync()` calls, but I'm not _sure_ that's safe to do, so I've shied away from keeping it in.

This has happened to me off and on; eventually it goes away on its own.

Does it go away if you reboot your machine? Now that we've got another repro, I'm wondering if that filesystem id sequence is per the lifetime of the filesystem, or per the uptime of the machine.
